src/parse-result.js

```diff
@@ -57,7 +57,7 @@ export function namespace(options) {
     }
   }
 
-  class SourceMap extends minim.Element {
+  class SourceMap extends minim.elements.Array {
```

Comment:
It's to prevent the following so that initialising `SourceMap` uses Array initialiser to handle refraction of JS primitive types:

```diff
diff --git a/src/parser.js b/src/parser.js
index 5e2bd20..639dc1c 100644
--- a/src/parser.js
+++ b/src/parser.js
@@ -65,7 +65,7 @@ export default class Parser {
 
   parse(done) {
     const {
-      Category, ParseResult, SourceMap,
+      Category, ParseResult, SourceMap, Number: NumberElement
     } = this.minim.elements;
     const swaggerParser = new SwaggerParser();
 
@@ -81,7 +81,12 @@ export default class Parser {
 
       if (err.mark) {
         this.result.first().attributes.set('sourceMap', [
-          new SourceMap([[err.mark.position, 1]]),
+          new SourceMap([
+            [
+              new NumberElement(err.mark.position),
+              new NumberElement(1),
+            ],
+          ]),
         ]);
       }
 
@@ -144,7 +149,10 @@ export default class Parser {
         const matches = err.message.match(/\$ref pointer "(.*?)"/);
 
         if (matches) {
-          location = [this.source.indexOf(matches[1]), matches[1].length];
+          location = [
+            new NumberElement(this.source.indexOf(matches[1])),
+            new NumberElement(matches[1].length),
+          ];
         }
 
         const annotation = this.createAnnotation(annotations.VALIDATION_ERROR,
@@ -250,7 +258,12 @@ export default class Parser {
           const position = err.problem_mark.pointer;
 
           annotation.attributes.set('sourceMap', [
-            new SourceMap([[position, 1]]),
+            new SourceMap([
+              [
+                new NumberElement(position),
+                new NumberElement(1),
+              ],
+            ]),
           ]);
         }
       }
@@ -1291,11 +1304,17 @@ export default class Parser {
   createSourceMap(element, path) {
     if (this.ast) {
       const SourceMap = this.minim.getElementClass('sourceMap');
+      const NumberElement = this.minim.getElementClass('number');
       const position = this.ast.getPosition(path);
 
       if (position && !isNaN(position.start) && !isNaN(position.end)) {
         element.attributes.set('sourceMap', [
-          new SourceMap([[position.start, position.end - position.start]]),
+          new SourceMap([
+            [
+              new NumberElement(position.start),
+              new NumberElement(position.end - position.start),
+            ],
+          ]),
         ]);
       }
```